api/include/opentelemetry/trace/tracer_provider.h

```diff
@@ -29,3 +29,27 @@ class TracerProvider
 };
 }  // namespace trace
 OPENTELEMETRY_END_NAMESPACE
+
+#if defined(OPENTELEMETRY_EXPORT)
+
+namespace std
+{
+
+//
+// Partial specialization of default_delete used by unique_ptr or shared_ptr.
+// This makes the delete of the type in unique_ptr/shared_ptr happening in the
+// DLL in which it is allocated.
+//
+template <>
+struct OPENTELEMETRY_EXPORT default_delete<OPENTELEMETRY_NAMESPACE::trace::TracerProvider>
```

Comment:
Should we have this definition only specific for `DLL` scenario ?

Reply:
This is removed now.